test/honeysql_postgres/postgres_test.cljc

```diff
@@ -65,6 +96,17 @@
                          :do-update-set {:fields [:phone :name]
                                          :where  [:= :user.active false]}}}))))
 
+(deftest filter-test
+  (is (= ["count(*) FILTER (WHERE NOT i BETWEEN ? AND ?) AS a" 3 5]
+         (sql/format (filter [(sql/call :count :*)(where [:not [:between :i 3 5]]) :a]))))
+
+  (is (= ["SELECT count(*) , count(*) FILTER (WHERE s.i < ?) AS foo, count(*) FILTER (WHERE s.i BETWEEN ? AND ?) AS bar FROM generate_series(1,10) AS s(i)" 5 3 10]
```

Comment:
There's an additional space after the first `SELECT count(*) ,` clause. 

Is that because of us [back-placing](https://github.com/nilenso/honeysql-postgres/pull/42/files#diff-f81b5b6904b08f9c50a240ecab26f660e4f13c113064b4519324a08f69ac62f8R121) the `,` if it's a select clause or has it always existed before?

Reply:
Looks like it gets added in before this expression is formatted. There's a space that gets added in between each different clause that's formatted, which is the cause of this. This has been the case for a few clauses. I'll dig deeper and open up an issue.

Reply:
Cool, let's open a separate issue and keep a track. Doesn't need to go in here since it didn't originate here.